server/sandbox_network.go

```diff
@@ -78,14 +79,27 @@ func (s *Server) networkStart(ctx context.Context, sb *sandbox.Sandbox) (podIPs
 				return nil, nil, fmt.Errorf("failed to get valid ip address for sandbox %s(%s)", sb.Name(), sb.ID())
 			}
 
-			err = s.hostportManager.Add(sb.ID(), &hostport.PodPortMapping{
-				Name:         sb.Name(),
-				PortMappings: sb.PortMappings(),
-				IP:           ip,
-				HostNetwork:  false,
-			}, "lo")
-			if err != nil {
-				return nil, nil, fmt.Errorf("failed to add hostport mapping for sandbox %s(%s): %v", sb.Name(), sb.ID(), err)
+			// use the corresponding IP family hostportManager for the IP
+			if utilnet.IsIPv6(ip) {
+				err = s.hostportManagerv6.Add(sb.ID(), &hostport.PodPortMapping{
+					Name:         sb.Name(),
+					PortMappings: sb.PortMappings(),
+					IP:           ip,
+					HostNetwork:  false,
+				}, "lo")
+				if err != nil {
+					return nil, nil, fmt.Errorf("failed to add hostport mapping for sandbox %s(%s): %v", sb.Name(), sb.ID(), err)
+				}
+			} else {
+				err = s.hostportManager.Add(sb.ID(), &hostport.PodPortMapping{
+					Name:         sb.Name(),
+					PortMappings: sb.PortMappings(),
+					IP:           ip,
+					HostNetwork:  false,
+				}, "lo")
+				if err != nil {
+					return nil, nil, fmt.Errorf("failed to add hostport mapping for sandbox %s(%s): %v", sb.Name(), sb.ID(), err)
+				}
```

Comment:
```
mapping := &hostport.PodPortMapping{
        ...
}
if utilnet.IsIPv6(ip) {
        err = s.hostportManagerv6.Add(sb.ID(), mapping, "lo")
} else {
        err = s.hostportManager.Add(sb.ID(), mapping, "lo")
}
if err != nil {
        ...
}
```

? Likewise in the remove case.

oh, also, you should be passing `""`, not `"lo"`. `HostportManager` used to require you to pass an interface name even if you didn't know the right interface, but now it doesn't and you're expected to pass `""` if you don't know the CNI plugin's bridge interface.

Reply:
done